pkg/cloud/api/path.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Path specifies a field in nested object. The type of the reference
```

Comment:
can we make this more json-path like for the slice index and maybe map key?

Reply:
the single character makes it easy to parse for someone using the path. Can you give an example of what you are thinking?